File: src/main/java/io/bindernews/thegrackle/cards/Cackle.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.powers.VulnerablePower;
import com.megacrit.cardcrawl.powers.WeakPower;
import io.bindernews.bnsts.CardNums;

public class Cackle extends BaseCard {
    public static final CardConfig C = new CardConfig("Cackle", CardType.SKILL);
    static final CardNums NUM = CardNums.builder()
            .cost(0)
            .magic(1).magicUpg(2)
            .build();

    public Cackle() {
        super(C, CardRarity.COMMON, CardTarget.ENEMY);
        NUM.init(this);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        addToBot(new ApplyPowerAction(m, p, new WeakPower(m, magicNumber, false), magicNumber));
        addToBot(new ApplyPowerAction(m, p, new VulnerablePower(m, magicNumber, false), magicNumber));
    }

    @Override
    public void upgrade() {
        NUM.upgrade(this, false);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/Cackle.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.powers.VulnerablePower
import com.megacrit.cardcrawl.powers.WeakPower
import io.bindernews.bnsts.CardVariables

class Cackle : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature) {
        addToBot(ApplyPowerAction(m, p, WeakPower(m, magicNumber, false), magicNumber))
        addToBot(ApplyPowerAction(m, p, VulnerablePower(m, magicNumber, false), magicNumber))
    }

    companion object {
        @JvmField val C = CardConfig("Cackle", CardType.SKILL, CardRarity.COMMON, CardTarget.ENEMY)
        val VARS = CardVariables.config { c ->
            c.cost(0, -1)
            c.magic(1, 2)
        }
    }
}